src/software/simulated_tests/serialized_to_nano_pb_simulator_adapter.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "software/proto/messages_robocup_ssl_wrapper.pb.h"
+#include "software/time/timestamp.h"
+#include "software/world/ball_state.h"
+#include "software/world/field.h"
+#include "software/world/robot_state.h"
+
+// We forward declare the Simulator because the simulator depends on NanoPb
+// PrimitiveSetMsg, which we will not expose to users of this class by only including
+// simulator.h in serialized_to_nano_pb_simulator_adapter.cpp
+class Simulator;
+/**
+ * This class adapts the Simulator's NanoPb protobuf interface to serialized protobufs, so
+ * that users of this class do not have to depend on NanoPb protobufs
+ */
+class SerializedToNanoPbSimulatorAdapter
```

Comment:
Can you clarify what the purpose of this wrapper class is? I see it's to prevent uses of the class from having to depend on NanoPb protobufs, but why? Seems like it's to maintain the nicer interface of providing "primitives" rather than bytes, because we can't have both the nanopb and google protobuf headers together.

I think we could avoid this with the conversion code @garethellis0 wrote (I don't see it in `master`, is it still part of your trajectory planner branch?). Then callers are free to provide / create google protobufs, they just need to call a conversion function to make it a nanopb protobuf before giving it to the simulator. Then we don't have to add another layer just to really abstract 1-2 functions.

Reply:
Since we cannot include google protobuf and nanopb in the same target (without some unsavoury namespacing), we cannot have a function that directly converts google protobuf to nanopb. @garethellis0 's solution was to convert `Primitive` to nanopb, bypassing google protobuf entirely. Since we've decided to delete the `Primitive` classes and use protobug `PrimitiveMsg`s directly, we need some way to avoid including google protobufs and nanopb in the same target, and this Adapter seemed like the only way

Reply:
Jon's correct, the adapter I wrote won't work here. Given that the only time we provide actual NanoPb messages to the simulator is during tests, why not change the simulator interface to only accept serialized primitives and add a helper function for the tests?

Reply:
@garethellis0 doesn't the adapter convert Google Protobufs to NanoPb?

Reply:
it's not possible to write a converter from Google Protobufs to NanoPb since we can't include two definitions of the same name. `ThreadedSimulator` uses the NanoPb Protobuf interface of `Simulator` so we can't easily remove that. I don't see how to do this

Reply:
Yeah, Jon is correct here. I think the simplest solution is to just use only `std::vector<uin8_t>` (ie. a serialized mesage) and then have a helper functions in for tests.

Reply:
`serialized_to_nano_pb_simulator_adapter` is a `test_only` target that lived in the `simulated_test` folder and is only used by the `SimulatedTestFixture`. I think this is better than changing the entire `StandaloneSimulator`->`ThreadedSimulator`->`Simulator` stack to only have `std::vector<uint8_t>` in their interfaces

Reply:
`serialized_to_nano_pb_simulator_adapter` still adds _another_ layer to the simulator though that will also have to be updated everytime functionality is added to the interface (eg. the upcoming changes to add/remove robots), which just makes it more of a pain to support IMO since we already have several layers of composition.

@jonathanlew is your main issue with changing the stack to use `std::vector<uint8_t>` interfaces due to the work of actually making the change, or because you feel it's a worse interface?

Reply:
`serialized_to_nano_pb_simulator_adapter` only needs to change if the `SimulatedTestFixture` requires more functionality to be exposed, not every time the `Simulator` gets more functionality. 

Yes, it requires a lot of work to make it work, but I def take issue with changing the entire stack to accommodate a quirk of the `SimulatedTestFixture`. Changing the stack to use `std::vector<uint8_t>` is a worse interface because it obfuscates the type that it represents and makes it possible to fail in more places. As it is currently, when running with the `StandaloneSimulator` binary, deserializing to nanopb proto can only fail in the nanopb listener. If we instead listened for `std::vector<uint8_t>` and passed that along `StandaloneSimulator`->`ThreadedSimulator`->`Simulator`, the failure can happen anywhere along that stack if we have to deserialize for whatever reason. Not to mention `StandaloneSimulatorTest`s, `ThreadedSimulatorTest`, and `SimulatorTest`s will have the overhead of serializing all primitives before passing them in

Reply:
Ok fair enough, I won't block this PR on this comment. Once this is merged and I finish adding more sim functionality we can see what it looks like.

Reply:
@jonathanlew Do you want to wait for #1662 to merge before this PR so we can remove the adapter, or put this through first?

Reply:
I'm gonna remove the adapter